examples/bert/run_pretraining.py

```diff
@@ -291,6 +301,53 @@ def train_step(self, data):
         self.nsp_accuracy.update_state(data["next_sentence_labels"], nsp_preds)
         return {m.name: m.result() for m in self.metrics}
 
+class AdamLRSchedule(keras.optimizers.schedules.LearningRateSchedule):
+    """
+    Implements a Learning Rate Scheduler for Adam with Linear Warmup and
+    Linear Decay for a fixed number of training steps instead of epochs.
+    """
+    
+    def __init__(self,
+            learning_rate,
+            num_warmup_steps,
+            num_train_steps
+        ):
+        self.threshold_learning_rate = learning_rate
+        self.warmup_steps = num_warmup_steps
+        self.train_steps = num_train_steps
+        
+    
+    def __call__(self, step):
+        """
+        Calculates learning rate with linear warmup and linear decay based
+        on the current step. Linear Warmup for the first `num_warmup_steps` and
+        Linear Decay for the steps post `num_warmup_steps` till
+        `num_train_steps`.
+        """
+        final_lr = tf.cast(self.threshold_learning_rate, dtype = tf.float32)
+        warmup = tf.cast(self.warmup_steps, dtype = tf.float32)
+        training = tf.cast(self.train_steps, dtype = tf.float32)
+        
+        is_warmup = step < warmup
+
+        #Linear Decay will be implemented if current step is not less than
+        #`num_warmup_steps`
+        learning_rate = tf.keras.optimizers.schedules.PolynomialDecay(
```

Comment:
I think this actually distracts from what's going on here. Can't we do something like this (though also converting to floats)?

```
if step < warmup_steps:
    return step / warmup_steps
else:
    return max(0.0, (train_steps - step) / (train_steps - warmup_steps))
```

Reply:
That is a better and a shorter way to do this, yeah. I was using this in the beginning but then I wondered if using `PolynomialDecay` would be the same or not, and just ended up using this. I'll change it and resolve this in a bit.